xla/service/gpu/llvm_gpu_backend/amdgpu_backend.cc

```diff
@@ -332,14 +315,94 @@ absl::StatusOr<std::vector<uint8_t>> EmitModuleToHsaco(
 
 // Links ROCm-Device-Libs into the given module if the module needs it.
 absl::Status LinkROCDLIfNecessary(llvm::Module* module,
```

Comment:
Can you create a test for this function? Something as simple as possible that tests whether the device libraries are getting linked or not?